internal/backend/backend.go

```diff
@@ -131,3 +138,90 @@ type resource struct {
 	ScannedAt        metav1.Time   `json:"scanned_at"`
 	DeletedAt        *metav1.Time  `json:"deleted_at,omitempty"`
 }
+
+type upsertFailures struct {
+	failures map[types.UID]*upsertFailure
+	oldest   *upsertFailure
+
+	failureHistogram *prometheus.HistogramVec
+	oldestFailure    *prometheus.GaugeVec
+}
+
+func newUpsertFailures() *upsertFailures {
+	u := &upsertFailures{
+		failures: make(map[types.UID]*upsertFailure),
+		oldest:   nil,
+
+		failureHistogram: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name: "backend_retries",
```

Comment:
Scanner metrics probably _should_ have a (configurable?) metrics namespace, e.g. `kubernetes_scanner`. We have dropped these in our internal apps because the new metrics pipeline automatically adds namespaces, but this won't be generally true for scanner users.

Reply:
Agreed, I'll give injecting this a shot.